src/lib/arch/aarch64/Instruction_decode.cc

```diff
@@ -314,25 +314,21 @@ void Instruction::decode() {
         } else if (op.access & cs_ac_type::CS_AC_READ) {
           sourceRegisters.push_back(csRegToRegister(op.sme_index.reg));
           operandsPending++;
-          sourceRegisterCount++;
         }
       }
       // Register that is base of index will always be a source operand
       sourceRegisters.push_back(csRegToRegister(op.sme_index.base));
       operandsPending++;
-      sourceRegisterCount++;
     } else if (op.type == ARM64_OP_REG_MRS) {
       int32_t sysRegTag = architecture_.getSystemRegisterTag(op.imm);
       if (sysRegTag == -1) {
         exceptionEncountered_ = true;
         exception_ = InstructionException::UnmappedSysReg;
         // Clear any registered operands
-        sourceRegisterCount = 0;
         destinationRegisterCount = 0;
```

Comment:
Wondering if we should just return at this stage rather than setting reg counts to zero

Reply:
Changed